packages/container/tests/registry_test.js

```diff
@@ -505,3 +506,170 @@ QUnit.test('A registry can be created with a deprecated `resolver` function inst
 
   equal(registry.resolve('foo:bar'), 'foo:bar-resolved', '`resolve` still calls the deprecated function');
 });
+
+if (isEnabled('ember-htmlbars-local-lookup')) {
+  // jscs:disable validateIndentation
+
+QUnit.test('resolver.expandLocalLookup is not required', function(assert) {
+  assert.expect(1);
+
+  var registry = new Registry({
+    resolver: { }
+  });
+
+  let result = registry.expandLocalLookup('foo:bar', {
+    source: 'baz:qux'
+  });
+
+  assert.equal(result, null);
+});
+
+QUnit.test('expandLocalLookup is called on the resolver if present', function(assert) {
+  assert.expect(4);
+
+  let resolver = {
+    expandLocalLookup(targetFullName, sourceFullName) {
+      assert.ok(true, 'expandLocalLookup is called on the resolver');
+      assert.equal(targetFullName, 'foo:bar', 'the targetFullName was passed through');
+      assert.equal(sourceFullName, 'baz:qux', 'the sourceFullName was passed through');
+
+      return 'foo:qux/bar';
+    }
+  };
+
+  var registry = new Registry({
+    resolver
+  });
+
+  let result = registry.expandLocalLookup('foo:bar', {
+    source: 'baz:qux'
+  });
+
+  assert.equal(result, 'foo:qux/bar');
+});
```

Comment:
We need to verify the prioritization given to the resolver, then to the fallback registry, like in https://github.com/rwjblue/ember.js/blob/local-lookup/packages/container/tests/registry_test.js#L316-L340

Something like this (I haven't actually tried to run this):

```
QUnit.test('`expandLocalLookup` is handled by the resolver, then by the fallback registry, if available', function(assert) {
  assert.expect(9);

  let fallback = {
    expandLocalLookup(targetFullName, sourceFullName) {
      assert.ok(true, 'expandLocalLookup is called on the resolver');
      assert.equal(targetFullName, 'foo:bar', 'the targetFullName was passed through');
      assert.equal(sourceFullName, 'baz:qux', 'the sourceFullName was passed through');

      return 'foo:qux/bar-fallback';
    }
  };

  let resolver = {
    expandLocalLookup(targetFullName, sourceFullName) {
      assert.ok(true, 'expandLocalLookup is called on the resolver');
      assert.equal(targetFullName, 'foo:bar', 'the targetFullName was passed through');
      assert.equal(sourceFullName, 'baz:qux', 'the sourceFullName was passed through');

      return 'foo:qux/bar-resolver';
    }
  };

  let registry = new Registry({
    fallback,
    resolver
  });

  let result = registry.expandLocalLookup('foo:bar', {
    source: 'baz:qux'
  });
  
  assert.equal(result, 'foo:qux/bar-resolver', 'handled by the resolver');
  
  registry.resolver = null;
  
  result = registry.expandLocalLookup('foo:bar', {
    source: 'baz:qux'
  });
  
  assert.equal(result, 'foo:qux/bar-fallback', 'handled by the fallback registry');
  
  registry.fallback = null;
  
  result = registry.expandLocalLookup('foo:bar', {
    source: 'baz:qux'
  });
  
  assert.equal(result, null, 'null is returned by default when no resolver or fallback registry is present');
});
```